### ISSUE TITLE
Button selected by default in range selector buttons

### ISSUE THREAD

**Issue by rsantet** [plotly/plotly.js#4709]
Hello all,

Right now, enabling the use of buttons in `layout.xaxis.rangeselector` makes it such that the button 'all' is always selected by default, no matter where it is positioned in the list of buttons. If the button 'all' is not present, the display is still rendered as if it was selected by default.

May we add a `default` (boolean) property inside a button to know which should be selected by default when first displaying the graph ? This could help to focus on a key point of the data, while allowing the user to zoom out by clicking the 'all' button or by double-clicking on the graph.
This property would be set to `false` by default, and if no buttons show a `true` value for this property, then the usual 'all' display shall be rendered.

Example :
<i>Usual display</i>
```
var layout = {
xaxis: {
            autorange: true,
            rangeselector: {
                buttons: [
                    {
                        count: 1,
                        label: '1m',
                        step: 'month',
                        stepmode: 'backward'
                    },
                    {
                        count: 6,
                        label: '6m',
                        step: 'month',
                        stepmode: 'backward'
                    },
                    {   
                        step: 'all'
                    }
                ]
            },
            rangeslider: {autorange: true},
            type: 'date'
        }
};
```

In the second case, when loading the page, we already have the 6 months zoom.
<i>6 months button selected by default</i>
```
var layout = {
        xaxis: {
            autorange: true,
            rangeselector: {
                buttons: [
                    {
                        count: 1,
                        label: '1m',
                        step: 'month',
                        stepmode: 'backward'
                    },
                    {
                        count: 6,
                        label: '6m',
                        step: 'month',
                        stepmode: 'backward',
                        default: true
                    },
                    {   
                        step: 'all'
                    }
                ]
            },
            rangeslider: {autorange: true},
            type: 'date'
        }
};
```

**Comment by jbhyperlite39** [comment]
Yes, please! I'd like to have the full archive of data loaded for exploration, but only have recent data displayed by default.  Makes bar charts easier to read.  Makes diagnosing recent errors in dataset easier to interpret.

**Comment by alexrblohm** [comment]
I like this suggestion, but I think you can do it kinda like this (sorry for the very weird syntax and stuff, this is taken from a larger app and adjusted for simplicity):
`from datetime import date
from dateutil.relativedelta import relativedelta

    fig.update_xaxes(rangeslider_visible=True
                     , range=[date.today() - relativedelta(months=+2), date.today()]
                     )
    fig.update_xaxes(
        rangeslider_visible=True,
        rangeselector=dict(
            buttons=list([
                dict(count=1, label="1m", step="month", stepmode="backward"),
                dict(count=2, label="2m", step="month", stepmode="backward"),
                dict(count=6, label="6m", step="month", stepmode="backward"),
                dict(count=1, label="YTD", step="year", stepmode="todate"),
                dict(count=1, label="1y", step="year", stepmode="backward"),
                dict(step="all")
            ])
        )
    )
`

**Comment by gvwilson** [comment]
Hi - we are trying to tidy up the stale issues and PRs in Plotly's public repositories so that we can focus on things that are still important to our community. Since this one has been sitting for several years, I'm going to close it; if it is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. Thanks for your help - @gvwilson

**Comment by Haptein** [comment]
I hope this is not too late to reopen it, a solution for this need would be super appreciated! This is still not possible in the latest version of plotly.py==5.23.0 (plotly.js 2.34.0).
